pkg/daemon/daemon.go

```diff
@@ -881,8 +882,8 @@ func checkFileContentsAndMode(filePath, expectedContent string, mode os.FileMode
 		glog.Errorf("could not read file: %q, error: %v", filePath, err)
 		return false
 	}
-	if strings.Compare(string(contents), expectedContent) != 0 {
-		glog.Errorf("content mismatch for file: %q; expected: %v; received: %v", filePath, expectedContent, string(contents))
+	if !bytes.Equal(contents, expectedContent) {
```

Comment:
this should now be much faster compared to string methods as it's directly implemented in assembly as well